aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,170 @@
+
+## Blazor components
+
+In SPA frameworks, apps are built with *components*. A component usually represents a piece of UI, such as a page, dialog, or data entry form. Components can be nested, reused, and shared between projects.
+
+In Blazor, a component is a .NET class. The class can either be written directly, as a C# class (*\*.cs*), or more commonly in the form of a Razor markup page (*\*.cshtml*).
+
+Razor, which has been around since 2010, is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with IntelliSense support. Here's an example of how to express a simple custom dialog component in a Razor file:
+
+```cshtml
+<div>
+    <h2>@Title</h2>
+    @RenderContent(Body)
+    <button onclick=@OnOK>OK</button>
+</div>
+
+@functions {
+    public string Title { get; set; }
+    public Content Body { get; set; }
+    public Action OnOK { get; set; }
+}
+```
+
+When this component is used elsewhere in the app, IntelliSense speeds development with syntax completion and parameter info.
+
+Many design patterns are possible with this simple foundation. This includes patterns recognizable in other SPA frameworks, such as stateful components, functional stateless components, and higher-order components.
+
+Components can be:
+
+* Nested.
+* Generated procedurally in code.
+* Shared via class libraries.
+* Unit tested without requiring a browser DOM.
+
+## Infrastructure
+
+Blazor offers the core facilities that most apps require, including:
+
+* Layouts
+* Routing
+* Dependency injection
+* Lazy loading (loading parts of the app on demand as a user navigates the app)
+* Unit testing
+
+As an important design point, all of these features are optional. When one of these features isn't used in an app, the implementation is stripped out of the app when published by the IL linker.
+
+Only a few low-level elements are baked into the framework. For example, routing and layouts aren't built-in. Routing and layouts are implemented in *user space*, code that an app developer can write without using internal APIs. These features can be replaced with different systems to suit the app's requirements. The current layouts prototype is implemented in only about 30 lines of C# code, so a developer can easily understand and replace it if desired.
+
+## Code sharing and .NET Standard
+
+The [.NET Standard](/dotnet/standard/net-standard) is a formal specification of .NET APIs that are intended to be available on all .NET implementations. Mono on WebAssembly supports `netstandard2.0` or higher. .NET Standard class libraries can be shared across backend server code and in browser-based apps.
```

Comment:
Why do you need ~backend~?

Reply:
It's just there to highlight the distinction about where the code resides. Not necessary ... striking it now. 🔪